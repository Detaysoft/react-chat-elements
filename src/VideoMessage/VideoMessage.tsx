import React from 'react'

import './VideoMessage.css'

import { FaCloudDownloadAlt, FaExclamationTriangle } from 'react-icons/fa'

import classNames from 'classnames'
import ProgressCircle from '../Circle/Circle'
import { IProgressOptions, IVideoMessageProps } from '../type'

const VideoMessage: React.FC<IVideoMessageProps> = props => {
  var progressOptions = {
    strokeWidth: 2.3,
    color: '#efe',
    trailColor: '#aaa',
    trailWidth: 1,
    step: (
      data: IProgressOptions,
      circle: {
        path: { setAttribute: (arg0: string, arg1: string) => void }
        value: () => number
        setText: (arg0: string | number) => void
      }
    ) => {
      circle.path.setAttribute('trail', (data.state !== undefined && data?.state?.color) || '')
      circle.path.setAttribute('trailwidth-width', (data.state !== undefined && data?.state?.width) || '')

      var value = Math.round(circle?.value() * 100)
      if (value === 0) circle?.setText('')
      else circle?.setText(value)
    },
  }

  const error = props?.data?.status && props?.data?.status.error === true
  const downloaded = props?.data?.status && props?.data?.status.download

  return (
    <div
      className={classNames('rce-mbox-video', {
        'padding-time': !props?.text,
      })}
    >
      <div
        className='rce-mbox-video--video'
        style={{
          ...(props?.data.width &&
            props?.data.height && {
              width: props.data.width,
              height: props.data.height,
            }),
        }}
      >
        {!downloaded && (
          <img
            src={props?.data.uri}
            alt={props?.data.alt}
            onClick={props.onOpen}
            onLoad={props.onLoad}
            onError={props.onPhotoError}
          />
        )}

        {downloaded && (
          <video controls controlsList={props.controlsList}>
            <source src={props?.data.videoURL} type='video/mp4' />
            Your browser does not support HTML video.
          </video>
        )}

        {error && (
          <div className='rce-mbox-video--video__block'>
            <span className='rce-mbox-video--video__block-item rce-mbox-video--error'>
              <FaExclamationTriangle />
            </span>
          </div>
        )}
        {!error && props?.data?.status && !downloaded && (
          <div className='rce-mbox-video--video__block'>
            {!props.data.status.click && (
              <button onClick={props.onDownload} className='rce-mbox-video--video__block-item rce-mbox-video--download'>
                <FaCloudDownloadAlt />
              </button>
            )}
            {typeof props.data.status.loading === 'number' && props.data.status.loading !== 0 && (
              <ProgressCircle
                animate={props.data.status.loading}
                className='rce-mbox-video--video__block-item'
                progressOptions={progressOptions}
              />
            )}
          </div>
        )}
      </div>
      {props?.text && <div className='rce-mbox-text'>{props.text}</div>}
    </div>
  )
}

export default VideoMessage
