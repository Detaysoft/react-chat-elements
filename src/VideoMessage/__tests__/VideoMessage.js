import React, { Component } from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import VideoMessage from '../VideoMessage'

describe('VideoMessage component', () => {
  it('should render without issues', () => {
    const component = shallow(<VideoMessage controlsList='' data={{}} />)

    expect(component.length).toBe(1)
    expect(toJson(component)).toMatchSnapshot()
  })
})
