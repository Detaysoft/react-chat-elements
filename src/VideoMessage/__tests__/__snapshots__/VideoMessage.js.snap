// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoMessage component should render without issues 1`] = `
<div
  className="rce-mbox-video padding-time"
>
  <div
    className="rce-mbox-video--video"
    style={Object {}}
  >
    <img />
  </div>
</div>
`;
